src/main/java/com/cryptomorin/xseries/XMaterial.java

```diff
@@ -1575,18 +1575,19 @@ public static XMaterial matchXMaterial(@Nonnull ItemStack item) {
     protected static Optional<XMaterial> matchDefinedXMaterial(@Nonnull String name, byte data) {
         // if (!Boolean.valueOf(Boolean.getBoolean(Boolean.TRUE.toString())).equals(Boolean.FALSE.booleanValue())) return null;
         Boolean duplicated = null;
+        boolean isAMap = name.equalsIgnoreCase("MAP");
 
         // Do basic number and boolean checks before accessing more complex enum stuff.
-        if (Data.ISFLAT || (data <= 0 && !(duplicated = isDuplicated(name)))) {
+        if (Data.ISFLAT || (!isAMap && data <= 0 && !(duplicated = isDuplicated(name)))) {
             Optional<XMaterial> xMaterial = getIfPresent(name);
             if (xMaterial.isPresent()) return xMaterial;
         }
-        // Usually flat versions wouldn't pass this point, but some special materials do.
 
+        // Usually flat versions wouldn't pass this point, but some special materials do.
         XMaterial oldXMaterial = requestOldXMaterial(name, data);
         if (oldXMaterial == null) {
             // Special case. Refer to FILLED_MAP for more info.
-            return data > 0 && name.endsWith("MAP") ? Optional.of(FILLED_MAP) : Optional.empty();
+            return (data >= 0 && isAMap) ? Optional.of(FILLED_MAP) : Optional.empty();
```

Comment:
Please also explain all these changes in this file.

Reply:
This change was made because in legacy versions,  the map is saved as "MAP" and at the moment of deserializing it will take the form of an empty map instead of a filled map (Ik, it's kind of confusing but i've already tested it and it is working as it should for legacy and newer versions), You can try yourself with the current version and will see the error i'm refering.

Reply:
Yeah, I'll need some time to test this. This material is a really special exception and it's really confusing to deal with like you mentioned.

Reply:
Can you move this to a new separate PR so it's easier to track? Thanks

Reply:
How can i do that?, sorry i kind of new in Git & GitHub.

Reply:
Open Git Bash in your fork's root directory and do this [this](https://stackoverflow.com/questions/39459467/)

Reply:
Done.